### PR TITLE
Add multi-region support

### PR DESCRIPTION
**Pull request by pmuens**
Adds multi-region awareness when creating deployments and archiving services.

Platform SDK PR --> https://github.com/serverless/platform-sdk/pull/13

### CODE REVIEW

File: lib/plugins/aws/deploy/lib/createStack.js [hunk 1]
```diff
@@ -36,7 +36,11 @@ module.exports = {
       params.NotificationARNs = this.serverless.service.provider.notificationArns;
     }
 
-    return this.createDeployment().then(() => this.provider.request(
+    const deploymentData = {
+      provider: this.serverless.service.provider.name,
+      region: this.serverless.service.provider.region,
+    };
+    return this.createDeployment(deploymentData).then(() => this.provider.request(
```
Comment (eahefnawy) [inline lib/plugins/aws/deploy/lib/createStack.js]: hmmm that doesn't seem right! the `createDeployment` call does not take any inputs!

File: lib/plugins/aws/deploy/lib/validateTemplate.js [hunk 2]
```diff
@@ -25,7 +25,11 @@ module.exports = {
       throw new Error(errorMessage);
     }).then(() => {
       if (!this.serverless.service.deployment || !this.serverless.service.deployment.deploymentId) {
-        return this.createDeployment().catch((error) => {
+        const deploymentData = {
+          provider: this.serverless.service.provider.name,
+          region: this.serverless.service.provider.region,
+        };
+        return this.createDeployment(deploymentData).catch((error) => {
           throw new Error(error);
```
Comment (eahefnawy) [inline lib/plugins/aws/deploy/lib/validateTemplate.js]: same here

**Comment by eahefnawy** [comment]
Another comment, we should merge this after both the platform sdk and backend changes are merged.

**Comment by pmuens** [comment]
This will be done in a different way. Closing...

**Pull request closed without merging**